File: 1800. Maximum Ascending Subarray Sum/javascript/main.js
/**
 * @param {number[]} nums
 * @return {number}
 */
const maxAscendingSum = function (nums) {
  if (nums.length === 0) return 0;  // Handle empty array case
  let result = nums[0];
  let sumTemp = nums[0];
  for (let index = 1; index < nums.length; index++) {
    if (nums[index] <= nums[index - 1]) {
      result = Math.max(result, sumTemp);
      sumTemp = nums[index];
    } else {
      sumTemp += nums[index];
    }
  }
  return Math.max(result, sumTemp);
};

console.log(maxAscendingSum([12,17,15,13,10,11,12]))

File: 1800. Maximum Ascending Subarray Sum/python/main.py
from typing import List
class Solution:
    def max_ascending_sum(self, nums: List[int]) -> int:
        if len(nums) ==0:
            return 0
        result = nums[0]
        temp_sum = nums[0]
        for i in range(1,len(nums)):
            if nums[i] <= nums[i-1]:
                result = max(result,temp_sum)
                temp_sum = nums[i]
            else:
                temp_sum+= nums[i]
                            
        return max(temp_sum,result)
    
    
if __name__ == "__main__":
    s = Solution()
    print(s.max_ascending_sum([10,20,30,5,10,50]))